silverback/_cli.py

```diff
@@ -142,3 +128,500 @@ def run(cli_ctx, account, runner_class, recorder, max_exceptions, path):
+
+
+@bots.command(name="list", section="Configuration Commands")
+def list_bots(client: ClusterClient):
+    """List all bots in a CLUSTER (Regardless of status)"""
+
+    if bot_names := list(client.bots):
+        click.echo(yaml.safe_dump(bot_names))
+
+    else:
+        click.secho("No bots in this cluster", bold=True, fg="red")
+
+
+@bots.command(name="info", section="Configuration Commands")
+@click.argument("bot_name", metavar="BOT")
+def bot_info(client: ClusterClient, bot_name: str):
+    """Get configuration information of a BOT in a CLUSTER"""
+
+    if not (bot := client.bots.get(bot_name)):
+        raise click.UsageError(f"Unknown bot '{bot_name}'.")
+
+    # NOTE: Skip machine `.id`, and we already know it is `.name`
+    click.echo(yaml.safe_dump(bot.model_dump(exclude={"id", "name", "environment"})))
+    if bot.environment:
+        click.echo("environment:")
+        click.echo(yaml.safe_dump([var.name for var in bot.environment]))
+
+
+@bots.command(name="update", section="Configuration Commands")
+@click.option("--new-name", "new_name")  # NOTE: No shorthand, because conflicts w/ `--network`
+@click.option("-i", "--image")
+@click.option("-n", "--network")
+@click.option("-a", "--account")
+@click.option("-g", "--group", "groups", multiple=True)
+@click.argument("name", metavar="BOT")
+def update_bot(
+    client: ClusterClient,
+    new_name: str | None,
+    image: str | None,
+    network: str | None,
+    account: str | None,
+    groups: list[str],
+    name: str,
+):
+    """Update configuration of BOT in CLUSTER
+
+    NOTE: Some configuration updates will trigger a redeploy"""
+
+    if new_name in client.bots:
+        raise click.UsageError(f"Cannot use name '{new_name}' to update bot '{name}'")
+
+    if not (bot := client.bots.get(name)):
+        raise click.UsageError(f"Unknown bot '{name}'.")
+
+    if new_name:
+        click.echo(f"Name:\n  old: {name}\n  new: {new_name}")
+
+    if network:
+        click.echo(f"Network:\n  old: {bot.network}\n  new: {network}")
+
+    redeploy_required = False
+    if image:
+        redeploy_required = True
+        click.echo(f"Image:\n  old: {bot.image}\n  new: {image}")
+
+    environment = list()
+    for vg_id in groups:
+        if "/" in vg_id:
+            vg_name, revision = vg_id.split("/")
+            vg = client.variable_groups[vg_name].get_revision(int(revision))
+
+        else:
+            vg = client.variable_groups[vg_id]
+
+        environment.append(vg)
+
+    set_environment = True
+
+    if len(environment) == 0 and bot.environment:
+        set_environment = click.confirm("Do you want to clear all environment variables?")
+
+    elif environment != bot.environment:
+        click.echo("old-environment:")
+        click.echo(yaml.safe_dump([var.name for var in bot.environment]))
+        click.echo("new-environment:")
+        click.echo(yaml.safe_dump([var for vg in environment for var in vg.variables]))
+
+    redeploy_required |= set_environment
```

Comment:
question: What is this?  I'm not familiar with `|=`.

Reply:
a = a | b